contracts/core/TransferProxy.sol

```diff
@@ -76,11 +80,18 @@ contract TransferProxy is
         external
         onlyAuthorized
     {
+        // Retrive current balance of token for the vault
+        uint exstingVaultBalance = ERC20(_tokenAddress).balanceOf(vaultAddress);
```

Comment:
"existingVaultBalance" vs "exstingVaultBalance"?

Reply:
This works for now. Though, did you have any thoughts re: using something like dYdX's TokenInteract?